### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by Ongy** [Ongy/waymonad#2]
This is a general roadmap issue to outline my ideas when which version will be released

- [x] 0.1 Basic usability -- The dev release
  - [x] `xdg-shell` support
  - [x] Some way to launch applications
  - [x] Keybinds
  - [x] Basic stackset, or what it will be (See #1 )
  - [x] Multiple workspaces
  - [x] Sensible focus follows "event" (switch workspace/new window etc.)
- [ ] 0.2 Events / Extensions -- The enthusiast release
  - [x] ~~Draft level~~ Managehook
  - [x] Basic event Bus/System
    - [x] Sensible core hooks
    - [x] Dynamic extensible hook
  - [ ] Fullscreen implementaton
    - [x] Manual user triggered
    - [ ] Application triggered fullscreen
  - [x] Mouse events (Focus follow)
  - [x] Extensible State
  - [x] Client Cursors
  - [x] Bar support. *Requires surface layers*
    - [x] Support X status bars with static struts (no toggle for now)
  - [x] Basic IPC (FUSE?)
    - [x] Outputs
       -- This allows for most of xrandr immediatly necessary
       - [x] Set Output mode
       - [x] Set output position
       - [x] read/write scale
    - [x] Seats/InputDevices
      - [x] Introspect
      - [x] Assign devices
      - [x] Set/read libinput options
    - [x] Workspaces
      - [x] Read current/total layout
      -  ~~Get layout cache~~ This is an output property
      - [x] Get current windows
    - [x] Dynamically attach/detacch shells (that are loaded) -- Code exists, needs some wlroots support
- [ ] 0.3 Implement wayland extension protocols/fun features
  - [ ] Filter mechanism for wl_globals
    - [x] Global registry
    - [x] Filter hook
    - [ ] Sane handling/setup/checking 
  - [ ] Input devices
      - [x] Drawing tablet *There's a Tablet protocol in wayland that has to be supported by wlroots, this maps into "legacy" devices*
        - [x] Tablet Tool
        - [x] Tablet Pad *This exists, but the mapping to legacy devices is lacking*
      - [ ] Touchscreen
        - [x] Basic implementation (I only got 1? touchpoint. Broken hardware, 2 in theory)
        - [ ] Do it properly, test multitouch (*hardware/contributions needed*)
  -  ~~Assigning input devices to seats~~ Got done for Fuse
  - [x] Wayland Output Scaling
  - [x] Gamma control
  - [x] screenshooter
  - [ ] server side decorations
    - [x] basic implementation
    - [ ] extension protocol implementation (blocked on protocol)
  - [x] submap
  - [x] Multiple instances of a single window
    - [x] Proxy view (original is alwasy right)
    - [x] MultiView (Focused window is right)
- [x] 0.4 stabalize layout type and implement some of them
        
    "Stabalised" by making code polymorphic over the type \o/
    - [x] Traditional XMonad style
    - [x] HLWM style
- [ ] 0.5 Look into making a few things that are core until here configurable
  - [ ] "slow" client strategy (for now forced scaling, maybe lock 
  - [ ] Enalbe other behaviour strategies on MultiViews
- [ ] 1.0 Release -- user release
- [ ] 1.? Generaly "Planed", but low priority, may land in official(ly supported) clients
  - [ ] Use hardware sensors to automatically set output rotation
  - [ ] Touch compositor interactions
    - [ ] N-finger swipe -> Change workspace
    - [ ] N-finger pinch -> ToggleFull
    - [ ] Better Mouse emulation
    - [ ] Your suggestions :)

**Comment by Ongy** [comment]
So I've done the 0.1 basics I wanted to have.

I think in theory waymonad should be usable at this point.
There's still a lot of convinience left to implement, so I don't recommend it, even if I titled it the dev release.

Just for reference, what I'll wait for to adopt it as experimental daily driver:
  * Bar support
  * xrandr/arandr like application **EDIT:** This is available over the fuse API now. Not atomic, but possible.
